processor/metricstransformprocessor/README.md

```diff
@@ -1,94 +1,155 @@
+      - action: {add_label, update_label, delete_label_value, toggle_scalar_data_type, aggregate_labels, aggregate_label_values}
+        # label specifies the label to operate on
+        label: <label>
+        # new_label specifies the updated name of the label; if action is add_label, new_label is required
+        new_label: <new_label>
+        # aggregated_values contains a list of label values that will be aggregated; if action is aggregate_label_values, aggregated_values is required
+        aggregated_values: [values...]
+        # new_value specifies the updated name of the label value; if action is add_label or aggregate_label_values, new_value is required
+        new_value: <new_value>
+        # label_value specifies the label value for which points should be deleted; if action is delete_label_value, label_value is required
+        label_value: <label_value>
+        # label_set contains a list of labels that will remain after aggregation; if action is aggregate_labels, label_set is required
+        label_set: [labels...]
+        # aggregation_type defines how excluded labels will be aggregated; if action is aggregate_labels or aggregate_label_values, aggregation_type is required
+        aggregation_type: {sum, mean, min, max}
+        # value_actions contain a list of operations that will be performed on the selected label
+        value_actions:
+            # value specifies the value to operate on
+          - value: <current_label_value>
+            # new_value specifies the updated value
+            new_value: <new_label_value>
 ```
 
 ## Examples
 
-### Insert New Metric
+### Create a new metric from an existing metric
 ```yaml
 # create host.cpu.utilization from host.cpu.usage
-metric_name: host.cpu.usage
+include: host.cpu.usage
 action: insert
 new_name: host.cpu.utilization
 operations:
   ...
 ```
 
-### Rename Metric
+### Rename metric
+```yaml
+# rename system.cpu.usage to system.cpu.usage_time
+include: system.cpu.usage
+action: update
+new_name: system.cpu.usage_time
+```
+
+### Rename multiple metrics using Substitution
 ```yaml
-# rename system.cpu.usage to cpu/usage_time
-metric_name: system.cpu.usage
+# rename all system.cpu metrics to system.processor.*
+include: ^system\.cpu\.(.*)$
 action: update
-new_name: cpu/usage_time
+new_name: system.processor.$1
+```
+
+### Add a label
+```yaml
+# for system.cpu.usage_time, add label `version` with value `opentelemetry collector vX.Y.Z` to all points
+include: system.cpu.usage
+action: update
+operations:
+  - action: add_label
+    new_label: version
+    new_value: opentelemetry collector {{version}}
+```
+
+### Add a label to multiple metrics
+```yaml
+# for all system metrics, add label `version` with value `opentelemetry collector vX.Y.Z` to all points
+include: ^system\.
+match_type: regexp
+action: update
+operations:
+  - action: add_label
+    new_label: version
+    new_value: opentelemetry collector {{version}}
```

Comment:
I might be a bit paranoid, but this case isn't part of the tests. Would you mind adding it?

Reply:
The application version is known at initialization time so this replacement is actually done then, when mapping the config to internal transforms. There is a test for this [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/processor/metricstransformprocessor/factory_test.go#L158).

Reply:
LGTM then